CRM/Chfunds/Upgrader.php

```diff
@@ -96,6 +98,132 @@ public function upgrade_1700() {
     return TRUE;
   }
 
+  public function upgrade_1800() {
+    $this->ctx->log->info('CRM-1578-[DMS] - Duplicates of CH Funds Showing Under Administer > Funds');
+    //Adding a new column 'parent_id' for dataTable `civicrm_option_value_ch`
+    $sql = "ALTER TABLE `civicrm_option_value_ch` ADD COLUMN `parent_id` INT NULL DEFAULT NULL COMMENT 'parent id of CH option value' AFTER is_enabled_in_ch";
+    CRM_Core_DAO::executeQuery($sql);
+    return TRUE;
+
+  }
+
+  public function upgrade_1810() {
+    $this->ctx->log->info('CRM-1578-[DMS] -Manage Historical CH Funds data');
+    //Get option Group ID value for 'ch_fund'
```

Comment:
this code can be simplified with a single API chained call

 ```suggestion
  // Get funds with duplicate names
  $optionValues = civicrm_api4('OptionGroup', 'get', [
    'select' => [
      'id',
    ],
    'where' => [
      ['name', '=', 'activity_type'],
    ],
    'limit' => 1,
    'chain' => [
      'funds' => [
        'OptionValue', 'get', [
          'select' => [
            'name', 
            'COUNT(name) AS count'
          ], 
          'groupBy' => [
            'name'
          ], 
          'having' => [
            ['count', '>', 1]
          ], 
          'where' => [
            ['option_group_id', '=', '$id']
          ]
        ]
      ],
    ],
  ]);

  if ( count($optionValues) > 0 && count($optionValues[0]['funds']) > 0 {
    // loop through $optionValues[0]['funds']
  }
```

Reply:
api test url: https://test2-dms.canadahelps.org/dms/api4#/explorer/OptionGroup/get?where=%5B%5B%22name%22,%22%3D%22,%22ch_fund%22%5D%5D&chain=%5B%5B%22name_me_0%22,%5B%22OptionValue%22,%22get%22,%22%7Bselect:%20%5Bname,%20COUNT(name)%20AS%20count%5D,%20groupBy:%20%5Bname%5D,%20having:%20%5B%5B'count',%20'%3E',%201%5D%5D,%20where:%20%5B%5Boption_group_id,%20%3D,%20$id%5D%5D%7D%22,%22%22%5D%5D%5D&limit=1&select=%5B%22id%22%5D